apps/music/test/marionette/loading_test.js

```diff
@@ -0,0 +1,78 @@
+/* global require, marionette, setup, suite, test */
+'use strict';
+
+var assert = require('assert');
+var Music = require('./lib/music.js');
+
+marionette('Music files loading', function() {
+  var apps = {};
+
+  var client = marionette.client({
+    prefs: {
+      'device.storage.enabled': true,
+      'device.storage.testing': true,
+      'device.storage.prompt.testing': true
+    },
+
+    settings: {
+      'lockscreen.enabled': false,
+      'ftu.manifestURL': null
+    },
+
+    apps: apps
+  });
+
+  var music;
+
+  setup(function() {
+    music = new Music(client);
+
+  });
+
+  suite('Detection of movies.', function () {
+    setup(function() {
+      client.fileManager.removeAllFiles();
+      client.fileManager.add([
+        {
+          // Regular Ogg Vorbis.
+          type: 'music',
+          filePath: 'apps/music/test-data/playlists/a.ogg'
+        },
+        {
+          // AAC file.
+          type: 'music',
+          filePath: 'apps/music/test-data/aac-tags.mp4'
+        },
+        {
+          // MPEG4 video file. Shouldn't load.
+          type: 'music',
+          filePath: 'test_media/Movies/gizmo2.mp4'
+        }
+      ]);
+    });
+
+    test('Check the mp4 video isn\'t loaded', function() {
+
+      music.launch();
+      music.waitForFirstTile();
+      music.waitFinishedScanning();
```

Comment:
I'm not convinced by the implementation of waitFinishedScanning(). It will return if the musicdb is not scanning. So if scanning has not started yet, it will also return immediately.  Ideally, music.launch() would inject an event handler that listens for the scanend event, and that event handler would set some kind of _initialScanComplete flag on the musicdb.  Then, waitFinishedScanning() can wait for that flag to be set.

I'd even be okay adding this line to shared/js/mediadb.js in the endscan() method at 1696:

  media.initialScanComplete = true;